### PR TITLE
SLES4SAP-15 new OS Edition Screen & Updates to SLES4SAP tests

### PR DESCRIPTION
**Pull request by alvarocarvajald**
1) Added test installation/sles4sap_product_installation to handle SLES4SAP-15's installation tests.
2) Updated installation/sles4sap_product_installation so it works with SLES4SAP-15.
3) Changed sles4sap/patterns so it works as a milestone test.
4) Changed sles4sap/saptune so tuned is started if it's not started by default
5) Updated handle_method() in lib/utils.pm so it doesn't type 2 usernames in a row (root and $testapi::username) when both ROOTONLY and DM_NEEDS_USERNAME are set

Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/563

### CODE REVIEW

**Comment by okurz** [comment]
Your commit messages look very nice and have good details but please phrase them in "imperative mood" as we commonly do, e.g. "Add test …" instead of "Added test …"

File: tests/installation/sles4sap_product_installation_mode.pm [hunk 1]
```diff
@@ -1,22 +1,31 @@
 #!/usr/bin/perl -w
-# G-Summary: Add SLES4SAP tests
-# G-Maintainer: Denis Zyuzin <dzyuzin@suse.com>
+# Summary: Handle "Choose Operation System Edition" screen for SLES4SAP installation flow
+# Maintainer: Alvaro Carvajal <acarvajal@suse.de>
```
Comment (okurz) [inline tests/installation/sles4sap_product_installation_mode.pm]: It seems this file is missing the copyright and license header. Could you please add one here as we have in other files?

File: tests/sles4sap/saptune.pm [hunk 2]
```diff
@@ -26,6 +26,10 @@ sub run {
 
     select_console 'root-console';
 
+    unless (tuned_is 'running') {
+        script_run "saptune daemon start";
```
Comment (okurz) [inline tests/sles4sap/saptune.pm]: That should not be allowed to fail, is it? I recommend by default to use `assert_script_run` or explicitly check the return value if you want to act on it differently

File: tests/installation/sles4sap_product_installation_mode.pm (hunk 1 above)
Comment (alvarocarvajald) [reply tests/installation/sles4sap_product_installation_mode.pm]: Sure

File: tests/sles4sap/saptune.pm (hunk 2 above)
Comment (alvarocarvajald) [reply tests/sles4sap/saptune.pm]: You're correct

**Comment by okurz** [comment]
* Please rewrite your commit messages as stated before, see https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history for help if needed

**Comment by okurz** [comment]
Please provide a link or screenshot to a local verification run.

**Comment by alvarocarvajald** [comment]
http://mango.suse.de/tests/91

Is that good enough? The tests didn't finish due to https://bugzilla.suse.com/show_bug.cgi?id=1059143

**Comment by alvarocarvajald** [comment]
Updated needles for this PR: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/568

**Comment by schlad** [comment]
@alvarocarvajald openQA project, as the upstream one, tries hard to follow certain rules while it comes to the commit messages. This is painful however helps all of us by having consistent git history. 

The same applies to the need of a verification run - this is needed so minimize the risk of breaking openQA testing badly. Again, I know it is painful, but helps all of us in a long run.

What worries me is that you have no option to get quick initial review. Could you contact your TL (Paul, right?) and make sure the team does provide you with needed support?
In QA SLE Nbg, I try to check if there are any old PRs (from my team members) not being reviewed and we as a team try to mitigate such situations. I would assume such approach could be good for your team.

I could also offer help myself - if you have an important PR, you could contact me directly, so I could see how the team could help you in getting quick initial review.

Is there anything else we qa sle nbg could support you better?

**Comment by okurz** [comment]
The verification job looks good enough for now. I don't really understand if the "first_boot" login issue isn't something specific to your changes. But I consider the code changes "safe enough" so merging :)
